hauler_all_the_things.sh

```diff
@@ -30,21 +30,32 @@ function info_ok { echo -e "$GREEN" "ok" "$NO_COLOR" ; }
 
 export PATH=$PATH:/usr/local/bin
 
+# Package Manager 
+if [[ -f /etc/debian_version ]]; then 
+  export PKG_MANAGER='apt-get';
+elif [[ -f /etc/redhat-release ]]; then
+  export PKG_MANAGER='yum';
+fi
+
 # el version
-if which rpm > /dev/null 2>&1 ; then export EL=$(rpm -q --queryformat '%{RELEASE}' rpm | grep -o "el[[:digit:]]" ) ; fi
+if which rpm > /dev/null 2>&1 ; then 
+  export EL=$(rpm -q --queryformat '%{RELEASE}' rpm | grep -o "el[[:digit:]]" ) ; 
+else
+  export EL="el8"
+fi
 
 # check for root
-if [ $(whoami) != "root" ] ; then fatal "please run $0 as root"; fi
+# if [ $(whoami) != "root" ] ; then fatal "please run $0 as root"; fi
```

Comment:
what is the thought behind this?

Reply:
Make the function `build` works with any user (not only root). in the Github runner, the mount point is own by user runner, if I launch the build then I have a permission error with root. 

I move this check inside `base` and `hauler_setup` functions.

Reply:
Let me take a look at the different functions to see if I can move it.

Reply:
just pushed a version that fixes this.